src/modules/storage/Storages.ts

```diff
@@ -20,27 +20,32 @@ export class Storages
     /**
      *  The instance of sqlite3
      */
-    protected db: sqlite.Database;
+    protected db: any;
+
+    /**
+     *  The database path
+     */
+    protected db_path: string;
 
     /**
      * Constructor
      * @param filename Valid values are filenames,
      * ":memory:" for an anonymous in-memory database and
      * an empty string for an anonymous disk-based database
-     * @param callback : If provided, this function will be called when
-     * the database was opened successfully or when an error occurred.
-     * The first argument is an error object.
      */
-    constructor (callback?: any)
+    constructor (filename: string)
+    {
+        this.db_path = filename;
+    }
+
+    /**
+     * Open the database
+     */
+    public open ()
     {
-        let filename: string = ".cache";
-        this.db = new sqlite.Database(filename,
+        this.db = new sqlite.Database(this.db_path,
```

Comment:
What if we create it only when we have a db instance?

Reply:
Do you want this? Please describe the points more specifically.

```
        if (this.db != null)
            this.db.close();
        this.db = new sqlite.Database(this.db_path,
            sqlite.OPEN_CREATE | sqlite.OPEN_READWRITE |
            sqlite.OPEN_SHAREDCACHE);
        this.db.configure("busyTimeout", 1000);
```